recipes-kernel/linux-firmware/linux-firmware_%.bbappend

```diff
@@ -0,0 +1,40 @@
+SRC_URI += "git://github.com/NXP/imx-firmware.git;protocol=https;destsuffix=imx-firmware;name=imx-firmware"
+SRCREV_imx-firmware = "484d38224fa2c26b8859a7bf20b7c4d49100f5bc"
+SRCREV_FORMAT = "default_imx-firmware"
+
+do_install_append() {
+    install -m 0644 ${WORKDIR}/imx-firmware/cyw-wifi-bt/*CYW43*/* ${D}${nonarch_base_libdir}/firmware/brcm
+}
+
+PACKAGES =+ "${PN}-bcm4359-pcie"
+
+FILES_${PN}-bcm4339 += " \
+    ${nonarch_base_libdir}/firmware/brcm/brcmfmac4339-sdio.txt \
+    ${sysconfdir}/firmware/BCM4335C0.ZP.hcd \
+"
+
+FILES_${PN}-bcm43430 += " \
+    ${nonarch_base_libdir}/firmware/brcm/brcmfmac43430-sdio.clm_blob \
+    ${nonarch_base_libdir}/firmware/brcm/brcmfmac43430-sdio.txt \
+    ${sysconfdir}/firmware/BCM43430A1.1DX.hcd \
+"
+
+FILES_${PN}-bcm43455 += " \
+    ${nonarch_base_libdir}/firmware/brcm/brcmfmac43455-sdio.clm_blob \
+    ${nonarch_base_libdir}/firmware/brcm/brcmfmac43455-sdio.txt \
+    ${sysconfdir}/firmware/BCM4345C0.1MW.hcd \
+"
+
+FILES_${PN}-bcm4356-pcie += " \
+    ${nonarch_base_libdir}/firmware/brcm/brcmfmac4356-pcie.clm_blob \
+    ${nonarch_base_libdir}/firmware/brcm/brcmfmac4356-pcie.txt \
+    ${sysconfdir}/firmware/BCM4354A2.1CX.hcd \
```

Comment:
Why are these firmware binaries distributed by NXP and are not included into _linux-firmware_ base recipe?

There is nothing wrong from technical implementation, but might rather bring additional licensing questions: which license should be used for which binaries loaded for the _same_ module.

Reply:
Not an expert here. We are redistributing the firmware from Murata, so as I understand it it is up to them to contribute it.

Reply:
> Not an expert here. We are redistributing the firmware from Murata,

Would it be possible to compare your Murata re-distribution license to the one available in _linux-firmware_ then? If they do match - then it is OK from licensing point of view to add those binaries under same terms.

If they do differ however, then I suggest you create a separate package for those re-distributed binaries and include License terms to it. This should ensure compliance and distribution policies.

> so as I understand it it is up to them to contribute it.

Have you tried to contact them for this? It might well be that they would add it to the _linux-firmware_ so there would be no need for NXP to re-distribute it and to keep this append here.

-- andrey

Reply:
> Would it be possible to compare your Murata re-distribution license to the one available in linux-firmware then? If they do match - then it is OK from licensing point of view to add those binaries under same terms.
>
>If they do differ however, then I suggest you create a separate package for those re-distributed binaries and include License terms to it. This should ensure compliance and distribution policies.

Okay, I was confused and the firmware is from Cypress (was Broadcom). We were working with Murata modules, which is how that got into the conversation. We are in fact supposed to be using our EULA for re-distribution, so these recipe changes are wrong. We really appreciate you pushing these questions :-)

We have switched away from this WiFi vendor, so what should I do next? Shall I go ahead and fix the licensing with a separate package, or should we just drop it?

> Have you tried to contact them for this? It might well be that they would add it to the linux-firmware so there would be no need for NXP to re-distribute it and to keep this append here.

Lauren says Cypress is not going to be interested in working on this.

Reply:
A specific package is better. So this is visible it is a specific firmware.